src/sage/plot/contour_plot.py

```diff
@@ -1659,6 +1658,14 @@ def region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol,
     from warnings import warn
     import numpy
 
+    plot_points = options['plot_points']
```

Comment:
Why `pop` here? Doesn't this change the contents of `options` before passing it on to those three other functions?

Reply:
here there is no pop :-) 

But, OK, there are pops for the other `@options` args, more precisely, for the `@option` with names which  in the previous version coincided with names of the function arguments.
Because such args merely serve as setting the default values, and then popped (by the decorator).

If you don't pop these args, you got errors such as 
```
   verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'incol'=red
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'outcol'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'bordercol'=blue
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderstyle'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'borderwidth'=None
    verbose 0 (163: primitive.py, options) WARNING: Ignoring option 'alpha'=1
    verbose 0 (163: primitive.py, options) 
    The allowed options for ContourPlot defined by a 150 x 150 data grid are:
```
as in a comment above.

In the case of `plot_points` there was a bug. `plot_points` are meant to be consumed by functions called in the body of  `region_plot`. Therefore we don't pop it.

Reply:
Ok thanks. Just wanted to be sure we weren't removing something from the dict that could have an effect on a plot three levels down.